src/mdtest.c

```diff
@@ -1340,10 +1353,9 @@ void summarize_results(int iterations, int print_time) {
 
     for (i = start; i < stop; i++) {
             min = max = all[i];
-            for (k=0; k < size; k++) {
-                for (j = 0; j < iterations; j++) {
-                    curr = all[(k*tableSize*iterations)
-                               + (j*tableSize) + i];
+            for (j = 0; j < iterations; j++) {
```

Comment:
Using better variable names than "i" and "j" and "k", like "iter" and "op" might avoid this kind of bug in the future.

Also, putting the "j*tableSize*size + k*tableSize + i" calculation into a small helper function like:

    int calc_allreduce_index(int index, int iter, int op)

(or whatever) and using it in the places where all[] is accessed would also avoid similar usage bugs.